pkg/kudoctl/kudoinit/crd/crds.go

```diff
@@ -96,13 +123,29 @@ func (c Initializer) verifyInstallation(client v1beta1.CustomResourceDefinitions
 	return nil
 }
 
-func (c Initializer) install(client v1beta1.CustomResourceDefinitionsGetter, crd *apiextv1beta1.CustomResourceDefinition) error {
+func (c Initializer) createOrUpdate(client v1beta1.CustomResourceDefinitionsGetter, crd *apiextv1beta1.CustomResourceDefinition) error {
 	_, err := client.CustomResourceDefinitions().Create(crd)
 	if kerrors.IsAlreadyExists(err) {
-		clog.V(4).Printf("crd %v already exists", crd.Name)
+		// We need to be careful here and never delete/recreate CRDs, we would delete
+		// all installed custom resources. We must have a correct update!
+		clog.V(4).Printf("crd %v already exists, try to update", crd.Name)
+
+		oldCrd, err := client.CustomResourceDefinitions().Get(crd.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get crd for update %s: %v", crd.Name, err)
+		}
+
+		crd.ResourceVersion = oldCrd.ResourceVersion
```

Comment:
Because we `update` the resource, this requires that the `resourceVersion` is the same as the one that we've retrieved in line 133. I guess the completely correct way here would be to calculate a patch and apply that, but as we're pretty sure that nobody modified our CRDs we just update them.